libcontainer/configs/config_linux.go

```diff
@@ -13,10 +13,10 @@ var (
 // different when user namespaces are enabled.
 func (c Config) HostUID(containerId int) (int, error) {
 	if c.Namespaces.Contains(NEWUSER) {
-		if c.UidMappings == nil {
+		if c.UIDMappings == nil {
```

Comment:
maybe we should use `len(c.UIDMappings) == 0 ` here?

Reply:
@eiffel-fl the ==nil should be removed. If it is nil the len is always 0. If it is not nil, the length can be 0 too.

So no need to keep both checks, we can just check for the len and that is enough.

Reply:
It seems like we didn't solve it but marked as resolved. Did I miss something here?

Reply:
Yes, maybe forgot at that time, thank you for your remind. This PR addressed too much problems in ONE PR(combine FD fields and rename *idMapping). Feel free to open an new PR to resolve this problem.

Reply:
https://github.com/opencontainers/runc/pull/3939#pullrequestreview-1538959937

Reply:
The new PR is here: #3947